### PR TITLE
Fix operator to run shoot on seeds

### PR DESCRIPTION
**Pull request by MichaelEischer**
**How to categorize this PR?**

/area control-plane
/kind bug

**What this PR does / why we need it**:

Backport several fixes to allow hosting a seed on a gardener-operator cluster.

### CODE REVIEW

**Comment by maboehm** [comment]
/hold 
we need to decide if we want to publish the helm chart, or keep the kyverno policy

File: charts/gardener/provider-local/templates/deployment.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ spec:
         networking.gardener.cloud/to-runtime-apiserver: allowed
         networking.gardener.cloud/to-dns: allowed
         networking.gardener.cloud/to-kind-network: allowed
+        networking.resources.gardener.cloud/to-all-istio-ingresses-istio-ingressgateway-tcp-9443: allowed
```
Comment (timebertt) [inline charts/gardener/provider-local/templates/deployment.yaml:35]: We should add notes to the commit messages, when the respective commits can be dropped again.

Comment (MichaelEischer) [reply charts/gardener/provider-local/templates/deployment.yaml:35]: I've added a remark in the commit message of the first two commits.

**Comment by MichaelEischer** [comment]
/hold cancel

We'll keep the Kyverno policy for now as we're only 4 Gardener version away from dropping the patch anyways.

**Pull request merged**